1_fetch.R

```diff
@@ -1,38 +1,250 @@
-                #as.data.frame() %>% 
+                as.data.frame() %>% 
                 mutate(HUC8 = p1_huc8_vec),
-              error = function(e){message(paste('error - No gages found in huc8', p1_huc8_vec))})},
+              error = function(e){
+                return(warning(e$message))
+                },
+              warning = function(w){
+                return(message(paste(w$message, 'huc8:', p1_huc8_vec)))
+              }
+              )},
   pattern = map(p1_huc8_vec)
   ),
 
-  ## Pulling site no from gauge sites to then query nwis and WQP with data retrieval
+  # Pulling site data retrieval using whatNWISsites(). This target is a sf object can be joined to nwis datasets below via site_no 
+  ## subsetting huc8 multipolygon to simplify join in get_NWIS_site_no() 
   tar_target(
+    p1_site_in_watersheds_sf,
+    get_NWIS_site_no(basin_huc08 = p1_huc08_df$huc8,
+                     lake_watershed_sf = p1_get_lakes_huc8_sf %>% select(HUC8, Shape), 
+                     crs = selected_crs)
+),
+
+tar_target(
+    p1_site_no,
+    {p1_site_in_watersheds_sf %>% pull(site_no)}
+),
+  
+ tar_target(
     p1_site_ids,
     {p1_nwis_sites %>% pull(site_no) %>% unique()}
-  ),
+),
+  
 # Download states shp
-tar_target(
+ tar_target(
   p1_download_states_shp,
   download_states_shp(url = states_download_url, 
                       out_path = '1_fetch/in/states_shp'),
   format = 'file'
 ),
 
-tar_target(
-  p1_states_sf,
-  st_read(file.path(p1_download_states_shp,'statesp010g.shp'), quiet = TRUE) %>%
-    filter(STATE_ABBR %in% c('CA',"NV",'UT','OR')) %>% 
-    st_transform(crs = st_crs(p1_lakes_sf)) %>% 
-    select(NAME,STATE_ABBR, geometry)
-)
+ tar_target(
+   p1_states_sf,
+   st_read(file.path(p1_download_states_shp,'statesp010g.shp'), quiet = TRUE) %>%
+     filter(STATE_ABBR %in% c('CA',"NV",'UT','OR')) %>% 
+     st_transform(crs = st_crs(p1_lakes_sf)) %>% 
+     select(NAME,STATE_ABBR, geometry)
+),
+
+  ###################
+  # NWIS Data Queries
+  
+  # SW
+  ## SW - dv
+  tar_target(
+    p1_nwis_dv_sw_data,
+    fetch_by_site_and_service(sites = p1_site_no,
+                              pcodes = p0_sw_params,
+                              service = 'dv',
+                              start_date = p0_start,
+                              end_date = p0_end)
+    ),
+
+  ## SW - iv
+  ## dv data is summarized from iv data, therefore any site with dv data will have iv data and vis versa
+  ## iv data is much heavier so we provided a filtered list to lighten the load of request
+  tar_target(
+    p1_nwis_iv_sw_data,
+    fetch_by_site_and_service(sites = unique(p1_nwis_dv_sw_data$site_no),
+                              pcodes = p0_sw_params,
+                              service = 'iv',
+                              start_date = p0_start,
+                              end_date = p0_end,
+                              incrementally = TRUE,
+                              split_num = 10)
```

Comment:
cool

Reply:
I think it may be useful to use some sort of chunking or branching to brake up other time intensive fetch steps